src/components/CalculatorDisplay.js

```diff
@@ -1,19 +1,79 @@
 import { Component } from 'react';
+import PropTypes from 'prop-types';
 
 class CalculatorDisplay extends Component {
+  static validateObject(obj) {
+    if (obj.next) {
+      return obj.next;
+    }
+    if (obj.total) {
+      return obj.total;
+    }
+    return 0;
+  }
+
   constructor(props) {
     super(props);
-    this.state = { display: '0' };
+    this.onDisplay = props.onDisplay;
+  }
+
+  handleKeyPress(e) {
+    const { onKeyEvent } = this.props;
+    switch (e.key) {
+      case '0':
+      case '1':
+      case '2':
+      case '3':
+      case '4':
+      case '5':
+      case '6':
+      case '7':
+      case '8':
+      case '9':
+      case '%':
+      case '+':
+      case '-':
+      case '=':
+      case '.':
+        onKeyEvent(e.key);
+        break;
+      case 'Escape':
+      case 'Cancel':
+        onKeyEvent('AC');
+        break;
+      case '/':
+        onKeyEvent('÷');
+        break;
+      case '*':
```

Comment:
- [ ] When doing multiplication the UI disappears and an error is thrown in the console, kindly fix this so all the operations are functional

![xx](https://user-images.githubusercontent.com/25227022/200122553-4822c2e5-b27f-482a-b09b-0ca338b4494f.gif)